### DISCUSSION TITLE
locale on ErrorBoundary

### DISCUSSION THREAD

**Discussion by cloud-walker** [remix-run/remix#9886] [Q&A]
Hi! I'm working on a multi locale application, and I'm trying to localize the 404 response from the root ErrorBoundary, but I don't see a way to pass the detected locale there, I'm missing something?

**Comment by sergiodxa** [comment]
If you know the root loader didn't failed, you can use `useRouteLoaderData<typeof loader>("root")` but remember the return type will have `| undefined` so you will have to do `loaderData?.locale` and probably also set a fallback value.

**Reply by cloud-walker** [reply]
hi @sergiodxa, I think to have tried that, but in my debugging session the hook always returned undefined, I think for the reason you are talking about (for example a 404 I think will always fail at the root level, no?)

**Comment by cloud-walker** [comment]
Just a second after opening the discussion, I remembered that I'm on a react-router based framework, the locale in my case is stored inside the pathname like that: `/<locale>/rest/of/the/pathname`.

So I tried to use the `useLocation` hook to extract the pathname inside the render of the root ErrorBoundary, and it seems to work 👯 

Dunno if this is the right take, but it seems to work 👍🏻
